sdk/include/mesh_dp.h

```diff
@@ -0,0 +1,352 @@
+     * Buffer size, or frame size, configured for the connection.
+     * This value is the maximum length of data the buffer may contain.
+     * It is calculated once before the connection is created and cannot be
+     * altered thereafter. The calculation is based on the payload type and
+     * payload parameters.
+     * For video payload, this value is the video frame size.
+     * For audio payload, this value is the audio packet size.
+     */
+    const size_t buf_size;
+
+} MeshConnection;
+
+/**
+ * Define configuration string field sizes
+ */
+#define MESH_SOCKET_PATH_SIZE   108
+#define MESH_IP_ADDRESS_SIZE    46
+
+/**
+ * Mesh configuration for Single node direct connection via memif
+ */
+typedef struct MeshConfig_Memif {
+    /**
+     * Memif socket path.
+     * Example: /run/mcm/mcm_memif_0.sock
+     */
+    char socket_path[MESH_SOCKET_PATH_SIZE];
+    /**
+     * Memif interface id.
+     * Default: 0
+     */
+    int interface_id;
+
+} MeshConfig_Memif;
+
+/**
+ * Mesh configuration for SMPTE ST2110-xx connection via Media Proxy
+ */
+typedef struct MeshConfig_ST2110 {
+    /**
+     * Remote IP address
+     */
+    char remote_ip_addr[MESH_IP_ADDRESS_SIZE];
+    /**
+     * Remote port
+     */
+    uint16_t remote_port;
+    /**
+     * Local IP address
+     */
+    char local_ip_addr[MESH_IP_ADDRESS_SIZE];
+    /**
+     * Local port
+     */
+    uint16_t local_port;
+    /**
+     * SMPTE ST2110-xx transport type.
+     * Must be aligned with the payload type.
+     * Any value of the MESH_CONN_TRANSPORT_ST2110_* constants.
+     */
+    int transport;
+#define MESH_CONN_TRANSPORT_ST2110_20 0 ///< SMPTE ST2110-20 Uncompressed Video transport
+#define MESH_CONN_TRANSPORT_ST2110_22 1 ///< SMPTE ST2110-22 Constant Bit-Rate Compressed Video transport
+#define MESH_CONN_TRANSPORT_ST2110_30 2 ///< SMPTE ST2110-30 Audio transport
+
+} MeshConfig_ST2110;
+
+/**
+ * Mesh configuration for RDMA connection via Media Proxy
+ */
+typedef struct MeshConfig_RDMA {
+            /**
+             * Remote IP address
+             */
+            char remote_ip_addr[MESH_IP_ADDRESS_SIZE];
+            /**
+             * Remote port
+             */
+            uint16_t remote_port;
+            /**
+             * Local IP address
+             */
+            char local_ip_addr[MESH_IP_ADDRESS_SIZE];
+            /**
+             * Local port
+             */
+            uint16_t local_port;
+
+} MeshConfig_RDMA;
+
+/**
+ * Mesh payload configuration for Video frames
+ */
+typedef struct MeshConfig_Video {
+    /**
+     * Video frame width in pixels.
+     */
+    int width;
+    /**
+     * Video frame height in pixels.
+     */
+    int height;
+    /**
+     * Video frames per second.
+     */
+    double fps;
+    /**
+     * Video frame pixel format.
+     * Any value of the MESH_VIDEO_PIXEL_FORMAT_* constants.
+     */
+    int pixel_format;
+#define MESH_VIDEO_PIXEL_FORMAT_NV12        0 ///< planar YUV 4:2:0, 12bpp
```

Comment:
The idea was to avoid introducing unnecessary type definitions for the user. Just use `int`. This concept is used in the FFmpeg code base. And the other reason to adopt that approach was to make the options visible along with the structure field definition.

These macros are only used as options of the structure field definition above them. Adding an `enum` will move their definitions far from the structure field, which will make it harder to understand by the user since they will need to scroll the code back and forth.

I agree, it's not an ideal approach, but I liked so much how similar structure definitions with macros bring clarity of the relationship between the field and the options in FFmpeg codebase. So I decided to adopt that.

Reference: https://github.com/FFmpeg/FFmpeg/blob/b9145fcab27da9e14c6929036d771afd1907a771/libavformat/avformat.h#L1302

Reply:
And what about the case when the same enums/defines are used in multiple places? Should we define them twice and try to not include header file in which they were defined previously? or define just ones and lose benefit of them being defined right above the field definition?

Reply:
I'd go with `enum` if the compiler doesn't allow you to do the below. **But it does**. GCC treats `enum` as an `int` and prints no warnings/errors even with `-Wall -Wenum_compare`.
```
enum MeshVideoPixelFormat {
    MESH_VIDEO_PIXEL_FORMAT_NV12        = 0, ///< planar YUV 4:2:0, 12bpp
    MESH_VIDEO_PIXEL_FORMAT_YUV422P     = 1, ///< planar YUV 4:2:2, 16bpp
    MESH_VIDEO_PIXEL_FORMAT_YUV422P10LE = 2, ///< planar YUV 4:2:2, 20bpp
    MESH_VIDEO_PIXEL_FORMAT_YUV444P10LE = 3, ///< planar YUV 4:4:4, 30bpp
    MESH_VIDEO_PIXEL_FORMAT_RGB8        = 4, ///< packed RGB 3:3:2,  8bpp
};

enum MeshVideoPixelFormat pixel_format;

...

pixel_format = 123;

Reply:
These macro constants have a four-level namespace prefix to make them look specifically related to the MCM library. Do you think the user will be defining something like `MESH_VIDEO_PIXEL_FORMAT_YUV422P10LE` in their code?

Reply:
ahh right, I forgot that only C++ prohibits this. Ok, type safety was my strongest argument. I would have chosen enum anyway, but it's mostly an opinion now. I don't like how struct filled with definitions looks, how it has broken indentation, but if you think that's better, We can do it this way

Reply:
Thank you. I'll collect all the feedback from reviewers to decide if we go with macros or refactor to `enum`.

Reply:
This discussion doesn't make much sense now because the maintainers agreed on switching to a JSON-based configuration in the next release.